### DISCUSSION TITLE
posts/how-to-write-a-vector

### DISCUSSION THREAD

**Discussion by giscus** [Locietta/blog-lost-pieces#3] [Announcements]
# posts/how-to-write-a-vector

变长数组容器，结合我现在对C++的一些理解来写写看。

https://blog.locietta.xyz/posts/how-to-write-a-vector

**Comment by morninghut** [comment]
果然不能自己实现的东西就不会完全理解，这样看来自己手敲实现 C++ 的 STL 还是挺有意义的嘛（跃跃欲逝）

> date: 2023-01-27

~~看来确实得猴年马月才能写好(x)~~

> 大概还得讨论realloc, iterator/ranges, trivally-copyable, 强异常安全之类的话题，结合一些C++20/23的语法。

~~所以读完这篇 blog 我就精通 C++ 了~~


总之写的棒棒[ ﾟ∀ﾟ]bb
